Docs updates -- Do not merge (#149)
Azure/azure-saas: docs/azure-saas-docs/content/en/resources/devops.md

| dotnet_version                  | The version of the .NET project to build                                                                                             |                                          |
| artifact_name                   | The name of the artifact file produced                                                                                               |                                          |
| app_service_name                | The name of the app service to deploy the artifact to                                                                                |                                          |
| app_service_resource_group_name | The name of the resource group the app service resides in                                                                            |                                          |
| project_build_path              | The path of the folder that the .csproj for the module is in                                                                         |                                          |

> | slot_name                       | The name of the deployment slot slot to create and deploy to on the app service (Used for override, recommended to keep the default) | pr-${{github.event.pull_request.number}} |

Reviewer: ```suggestion
| slot_name                       | The name of the deployment slot  to create and deploy to on the app service (Used for override, recommended to keep the default) | pr-${{github.event.pull_request.number}} |
```

#### Secrets

- `AZURE_CREDENTIALS`

>   - The devops pipeline must contain azure credentials for a service principal that has an appropriate access level to create a new app service slot and deploy to it. See [this](https://docs.microsoft.com/en-us/azure/developer/github/connect-from-azure?tabs=azure-portal%2Cwindows) document for setting up these credentials in your own repo. 

Reviewer: ```suggestion
  - The CI/CD pipeline must contain Azure credentials for a service principal that has an appropriate access level to create a new Azure App Service slot and deploy to it. See [this](https://docs.microsoft.com/azure/developer/github/connect-from-azure?tabs=azure-portal%2Cwindows) document for setting up these credentials in your own repo. 
```

#### Job Breakdown
There are 3 jobs within this workflow: `build`, `create-deployment-slot`, and `deploy-to-slot`. Here is a high level overview of what each job does.

1. `build`
   1. Run the .NET restore, build, and publish commands on the project. Project is determined by the `project_build_path` input variable.
   2. Publishes built .NET artifact to GitHub artifacts. Artifact is named with the string provided in the `artifact_name` input variable.

2. `create-deployment-slot`
   1. Logs into the Azure CLI with [credentials provided](https://docs.microsoft.com/en-us/azure/developer/github/connect-from-azure?tabs=azure-portal%2Cwindows#create-a-service-principal-and-add-it-as-a-github-secret) in the `AZURE_CREDENTIALS` secret.

>    2. Runs Azure CLI command to create a new deployment slot in the app service provided in the `app_service_name` input variable. Slot name is provided via the `slot_name` input variable.

Reviewer: ```suggestion
   2. Runs Azure CLI command to create a new deployment slot in the Azure App Service provided in the `app_service_name` input variable. Slot name is provided via the `slot_name` input variable.
```

- `pr-merge-saas-application.yml`
- `pr-merge-saas-permissions.yml`
- `pr-merge-saas-signupadministration.yml`

Triggered On: `Pull request close`

#### Input Variables

| Input Name                      | Description                                                                                                         | Default                                  |
|---------------------------------|---------------------------------------------------------------------------------------------------------------------|------------------------------------------|
| app_service_name                | The name of the app service the slot is deployed to                                                                 |                                          |
| app_service_resource_group_name | The name of the resource group the app service resides in                                                           |                                          |
| slot_name                       | The name of the deployment slot the staged code is deployed to (Used for override, recommended to keep the default) | pr-${{github.event.pull_request.number}} |

#### Secrets

- `AZURE_CREDENTIALS`

>   - The devops pipeline must contain azure credentials for a service principal that has an appropriate access level to preform the swap operation on the app service in question. See [this](https://docs.microsoft.com/en-us/azure/developer/github/connect-from-azure?tabs=azure-portal%2Cwindows) document for setting up these credentials in your own repo. 

Reviewer: ```suggestion
  - The CI/CD pipeline must contain Azure credentials for a service principal that has an appropriate access level to perform the swap operation on the Azure App Service in question. See [this](https://docs.microsoft.com/en-us/azure/developer/github/connect-from-azure?tabs=azure-portal%2Cwindows) document for setting up these credentials in your own repo. 
```

#### Job Breakdown

There are 2 main jobs within this workflow: `swap-slot` and `delete-slot`

1. `swap-slot` - *This job only runs if the PR is closed AND merged. This job does not run if the PR is closed without merging.*
   1. Logs into the Azure CLI with [credentials provided](https://docs.microsoft.com/en-us/azure/developer/github/connect-from-azure?tabs=azure-portal%2Cwindows#create-a-service-principal-and-add-it-as-a-github-secret) in the `AZURE_CREDENTIALS` secret.

>    2. Runs the Azure CLI command to swap the deployment slot named in the `slot_name` input variable with the production slot on the app service named in the `app_service_name` input variable. 

Reviewer: ```suggestion
   2. Runs the Azure CLI command to swap the deployment slot named in the `slot_name` input variable with the production slot on the Azure App Service named in the `app_service_name` input variable. 
```

2. `delete-slot` - This job will only run if the preceding step runs and succeeds. If the PR is closed without merging, the previous step will be skipped but this will still run.
   1. Logs into the Azure CLI with [credentials provided](https://docs.microsoft.com/en-us/azure/developer/github/connect-from-azure?tabs=azure-portal%2Cwindows#create-a-service-principal-and-add-it-as-a-github-secret) in the `AZURE_CREDENTIALS` secret.

>    2. Runs the Azure CLI command to delete the deployment slot named in the `slot_name` input variable with the production slot on the app service named in the `app_service_name` input variable. 

Reviewer: ```suggestion
   2. Runs the Azure CLI command to delete the deployment slot named in the `slot_name` input variable with the production slot on the Azure App Service named in the `app_service_name` input variable. 
```
